File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/DisableIndex.py
from ravendb import DisableIndexOperation
from ravendb.documents.operations.definitions import VoidMaintenanceOperation

from examples_base import ExampleBase


class Foo:
    # region syntax
    class DisableIndexOperation(VoidMaintenanceOperation):
        def __init__(self, index_name: str, cluster_wide: bool = False): ...

    # endregion


class DisableIndex(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_disable_index(self):
        with self.embedded_server.get_document_store("DisableIndex") as store:
            self.add_index_orders_totals(store)
            # region disable_1
            # Define the disable index operation
            # Use this args set to disable on a single node
            disable_index_op = DisableIndexOperation("Orders/Totals")

            # Execute the operation by passing it to maintenance.send
            store.maintenance.send(disable_index_op)

            # At this point, the index is disabled only on the 'preferred node'
            # New data will not be indexed on this node only
            # endregion

            # region disable_2
            # Define the disable index operation
            # Pass 'True' to disable the index on all nodes in the database-group
            disable_index_op = DisableIndexOperation("Orders/Totals", True)

            # Execute the operation by passing it to maintenance.send
            store.maintenance.send(disable_index_op)

            # At this point, the index is disabled on ALL nodes
            # New data will not be indexed
            # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/DisableIndex.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Operations\Indexes\DisableIndexOperation;

class DisableIndex
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region disable_1
            // Define the disable index operation
            // Use this overload to disable on a single node
            $disableIndexOp = new DisableIndexOperation("Orders/Totals");

            // Execute the operation by passing it to Maintenance.Send
            $store->maintenance()->send($disableIndexOp);

            // At this point, the index is disabled only on the 'preferred node'
            // New data will not be indexed on this node only
            # endregion

        } finally {
            $store->close();
        }

        $store = new DocumentStore();
        try {
            # region disable_2
            // Define the disable index operation
            // Pass 'true' to disable the index on all nodes in the database-group
            $disableIndexOp = new DisableIndexOperation("Orders/Totals", true);

            // Execute the operation by passing it to Maintenance.Send
            $store->maintenance()->send($disableIndexOp);

            // At this point, the index is disabled on ALL nodes
            // New data will not be indexed
            # endregion
        } finally {
            $store->close();
        }
    }

}

/*
interface IFoo
{
    # region syntax
    DisableIndexOperation(?string $indexName, bool $clusterWide = false)
    # endregion
}
*/
